docker/images/n8n-custom/Dockerfile

```diff
@@ -1,55 +1,34 @@
-# 1. Create an image to build n8n
-FROM node:16-alpine as builder
-
-# Update everything and install needed dependencies
-USER root
+ARG NODE_VERSION=16
 
-# Install all needed dependencies
-RUN apk --update add --virtual build-dependencies python3 build-base ca-certificates git && \
-	npm_config_user=root npm install -g npm@latest run-script-os turbo
+# 1. Create an image to build n8n
+FROM n8nio/base:${NODE_VERSION} as builder
 
-WORKDIR /data
+RUN \
+	apk --no-cache add git && \
+	npm install -g run-script-os turbo
 
-COPY turbo.json .
-COPY package.json .
-COPY package-lock.json .
-COPY packages/cli/ ./packages/cli/
-COPY packages/core/ ./packages/core/
-COPY packages/design-system/ ./packages/design-system/
-COPY packages/editor-ui/ ./packages/editor-ui/
-COPY packages/nodes-base/ ./packages/nodes-base/
-COPY packages/workflow/ ./packages/workflow/
-RUN rm -rf node_modules packages/*/node_modules packages/*/dist
+COPY turbo.json package.json package-lock.json ./
+COPY packages ./packages
 
+RUN chown -R node:node .
 RUN npm config set legacy-peer-deps true
-RUN npm install --loglevel notice
-RUN npm run build
-
-
-# 2. Start with a new clean image with just the code that is needed to run n8n
-FROM node:16-alpine
 
-USER root
+USER node
 
-RUN apk add --update graphicsmagick tzdata tini su-exec git
+RUN \
+	npm ci && \
```

Comment:
any value in running `npm ci` separately to cache the dependencies in a layer of their own for the times we only change source code? I'm assuming `npm run build` generates changes on every source code change.

Reply:
The issue here is that `npm ci` is installing dev-dependencies as well, which are needed for the `npm run build`. These are later deleted by the `npm prune`. 
If we split this up, we end up with a significantly larger image, as the layer with `npm ci` contains all the dev-dependencies, despite us not needing them after the build.

I plan to optimize these layers in the future to reduce the image sizes even further, but none of those plans fit into the scope of this PR IMO.

Reply:
makes sense :+1: thanks